src/fastmeteo/data/arpege.py

```diff
@@ -0,0 +1,113 @@
+import tempfile
+from datetime import datetime, timedelta, timezone
+from io import BytesIO
+from pathlib import Path
+from typing import Any
+
+import httpx
+import numpy as np
+import pandas as pd
+import xarray as xr
+from tqdm.auto import tqdm
+
+from .. import aero
+from ..core.grid import Grid
+
+tempdir = Path(tempfile.gettempdir())
+
+bare_url = "https://object.data.gouv.fr/meteofrance-pnt/pnt/"
+
+
+def download_with_progress(url: str) -> BytesIO:
+    with httpx.stream("GET", url) as r:
+        total_size = int(r.headers.get("Content-Length", 0))
+        buffer = BytesIO()
+        with tqdm(
+            total=total_size, unit="B", unit_scale=True, desc=url.split("/")[-1]
+        ) as progress_bar:
+            for chunk in r.iter_bytes():
+                buffer.write(chunk)
+                progress_bar.update(len(chunk))
+        buffer.seek(0)
+        return buffer
+
+
+class Arpege(Grid):
+    """
+    Class to handle the ARPEGE data.
+    """
+
+    def __init__(self, local_store: str) -> None:
+        self.local_store = local_store
+        self.model = "ARPEGE"
+        self.grib = "025"
+        self.package = "IP1"
+        self.time_range = "000H024H"
```

Comment:
On 01, 00h-12h, 13h-24h, 25h-36h, 37h-48h, (1h step) 49h-60h, 61h-72h, 73h-84h, 85h-96h et 97h-102h (3h steps) are the available horizons

On 025, they are available as 0 to 24, 25 to 48, 49 to 72 and 73 to 102 (4 files)